dot/core/interface.go

```diff
@@ -41,8 +41,8 @@ type BlockState interface {
 	GetSlotForBlock(common.Hash) (uint64, error)
 	GetFinalisedHeader(uint64, uint64) (*types.Header, error)
 	GetFinalisedHash(uint64, uint64) (common.Hash, error)
-	RegisterImportedChannel(ch chan<- *types.Block) (byte, error)
-	UnregisterImportedChannel(id byte)
+	GetImportedBlockNotifierChannel() (chan *types.Block, error)
+	FreeImportedBlockNotifierChannel(ch chan *types.Block)
```

Comment:
```suggestion
	GetImportedBlockNotifierChannel() (<-chan *types.Block, error)
	FreeImportedBlockNotifierChannel(ch <-chan *types.Block)
```

update to receive-only channel

Reply:
I'm struggling with this, see general comment...